File: examples/start_building/v2/python/your_first_transaction.py
import json
import time
import base64
from algosdk import account

from algosdk.v2client import algod
from algosdk import transaction


def getting_started_example():
	algod_address = "http://localhost:4001"
	algod_token = "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
	algod_client = algod.AlgodClient(algod_token, algod_address)

    # Generate new account for this transaction
	secret_key, my_address = account.generate_account()
    
	print("My address: {}".format(my_address))

    # Check your balance. It should be 0 microAlgos

	account_info = algod_client.account_info(my_address)
	print("Account balance: {} microAlgos".format(account_info.get('amount')) + "\n")

    # Fund the created account
	print('Fund the created account using testnet faucet: \n https://dispenser.testnet.aws.algodev.network/?account=' + format(my_address))

	completed = ""
	while completed.lower() != 'yes':
		completed = input("Type 'yes' once you funded the account: ");

	print('Fund transfer in process...')
    # Wait for the faucet to transfer funds
	time.sleep(10)	

	account_info = algod_client.account_info(my_address)
	print("Account balance: {} microAlgos".format(account_info.get('amount')))

	# build transaction
	params = algod_client.suggested_params()
	# comment out the next two (2) lines to use suggested fees
	params.flat_fee = True
	params.fee = 1000
	receiver = "HZ57J3K46JIJXILONBBZOHX6BKPXEM2VVXNRFSUED6DKFD5ZD24PMJ3MVA"
	note = "Hello World".encode()
	amount = 1000000
	unsigned_txn = transaction.PaymentTxn(my_address, params, receiver, amount, None, note)

	# sign transaction
	signed_txn = unsigned_txn.sign(secret_key)
	txid = algod_client.send_transaction(signed_txn)
	print("Signed transaction with txID: {}".format(txid))

    # wait for confirmation	
	try:
		confirmed_txn = transaction.wait_for_confirmation(algod_client, txid, 4)  
	except Exception as err:
		print(err)
		return

	print("Transaction information: {}".format(
		json.dumps(confirmed_txn, indent=4)))
	print("Decoded note: {}".format(base64.b64decode(
		confirmed_txn["txn"]["txn"]["note"]).decode()))
	print("Starting Account balance: {} microAlgos".format(account_info.get('amount')) )
	print("Amount transfered: {} microAlgos".format(amount) )    
	print("Fee: {} microAlgos".format(params.fee) ) 
	account_info = algod_client.account_info(my_address)
	print("Final Account balance: {} microAlgos".format(account_info.get('amount')) + "\n")


getting_started_example()

File: examples/start_building/v2/go/yourFirstTransaction.go
package main

import (
	"context"
	json "encoding/json"
	// "errors"
	"fmt"
	// "strings"
	"github.com/algorand/go-algorand-sdk/future"
	"github.com/algorand/go-algorand-sdk/client/v2/algod"
	// "github.com/algorand/go-algorand-sdk/client/v2/common/models"
	"github.com/algorand/go-algorand-sdk/crypto"
	"github.com/algorand/go-algorand-sdk/mnemonic"
	"github.com/algorand/go-algorand-sdk/transaction"
)



// TODO: insert aditional utility functions here

func main() {
	// Create account
	account := crypto.GenerateAccount()
	passphrase, err := mnemonic.FromPrivateKey(account.PrivateKey)
	myAddress := account.Address.String()

	if err != nil {
		fmt.Printf("Error creating transaction: %s\n", err)
	} else {
		fmt.Printf("My address: %s\n", myAddress)
		fmt.Printf("My passphrase: %s\n", passphrase)
		fmt.Println("--> Copy down your address and passpharse for future use.")
		fmt.Println("--> Once secured, press ENTER key to continue...")
		fmt.Scanln()
	}

	// Fund account
	fmt.Println("Fund the created account using testnet faucet:\n--> https://dispenser.testnet.aws.algodev.network?account=" + myAddress)
	fmt.Println("--> Once funded, press ENTER key to continue...")
	fmt.Scanln()

	// instantiate algod client to Algorand Sandbox
	const algodAddress = "http://localhost:4001"
	const algodToken = "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"

	algodClient, err := algod.MakeClient(algodAddress, algodToken)
	if err != nil {
		fmt.Printf("Issue with creating algod client: %s\n", err)
		return
	}

	//Check account balance
	fmt.Printf("My address: %s\n", myAddress)

	accountInfo, err := algodClient.AccountInformation(myAddress).Do(context.Background())
	if err != nil {
		fmt.Printf("Error getting account info: %s\n", err)
		return
	}
	var startingAmount uint64 = accountInfo.Amount
	fmt.Printf("Account balance: %d microAlgos\n", accountInfo.Amount)
	fmt.Println("--> Ensure balance greater than 0, press ENTER key to continue...")
	fmt.Scanln()

	// Construct the transaction
	txParams, err := algodClient.SuggestedParams().Do(context.Background())
	if err != nil {
		fmt.Printf("Error getting suggested tx params: %s\n", err)
		return
	}
	fromAddr := myAddress
	toAddr := "GD64YIY3TWGDMCNPP553DZPPR6LDUSFQOIJVFDPPXWEG3FVOJCCDBBHU5A"
	// close to dispenser
	closeToAddr := "HZ57J3K46JIJXILONBBZOHX6BKPXEM2VVXNRFSUED6DKFD5ZD24PMJ3MVA"
	var amount uint64 = 1000000
	var minFee uint64 = 1000
	note := []byte("DevPortal - My First Transaction with Go SDK")
	genID := txParams.GenesisID
	genHash := txParams.GenesisHash
	firstValidRound := uint64(txParams.FirstRoundValid)
	lastValidRound := uint64(txParams.LastRoundValid)
	txn, err := transaction.MakePaymentTxnWithFlatFee(fromAddr, toAddr, minFee, amount, firstValidRound, lastValidRound, note, closeToAddr, genID, genHash)
	if err != nil {
		fmt.Printf("Error creating transaction: %s\n", err)
		return
	}

	// Sign the transaction
	txID, signedTxn, err := crypto.SignTransaction(account.PrivateKey, txn)
	if err != nil {
		fmt.Printf("Failed to sign transaction: %s\n", err)
		return
	}
	fmt.Printf("Signed txid: %s\n", txID)

	// Submit the transaction
	sendResponse, err := algodClient.SendRawTransaction(signedTxn).Do(context.Background())
	if err != nil {
		fmt.Printf("failed to send transaction: %s\n", err)
		return
	}
	fmt.Printf("Submitted transaction %s\n", sendResponse)

	// Wait for confirmation
	confirmedTxn, err := future.WaitForConfirmation(algodClient, txID, 4)
	if err != nil {
		fmt.Printf("Error wating for confirmation on txID: %s\n", txID)
		return
	}

	// Get the completed Transaction
	txnJSON, err := json.MarshalIndent(confirmedTxn.Transaction.Txn, "", "\t")
	if err != nil {
		fmt.Printf("Can not marshall txn data: %s\n", err)
	}
	fmt.Printf("Transaction information: %s\n", txnJSON)
	fmt.Printf("Decoded note: %s\n", string(confirmedTxn.Transaction.Txn.Note))
	fmt.Printf("Amount sent: %d microAlgos\n", confirmedTxn.Transaction.Txn.Amount)
	fmt.Printf("Fee: %d microAlgos\n", confirmedTxn.Transaction.Txn.Fee)	
	amountAndFee := uint64 (confirmedTxn.Transaction.Txn.Amount + confirmedTxn.Transaction.Txn.Fee)
	fmt.Printf("Close to Amount: %d microAlgos\n", startingAmount - amountAndFee)		
	fmt.Printf("Decoded note: %s\n", string(confirmedTxn.Transaction.Txn.Note))

	// TODO: insert additional codeblocks here
}
